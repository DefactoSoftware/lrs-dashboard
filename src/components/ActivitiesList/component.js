import React from 'react';
import Activity from '../Activity/component';
import CSS from './style.css';

export default ({ activities })=> {
  return (
    <ul className={CSS.ActivitiesList}>
      {activities.map(activity => (
        <li className={CSS.item} key={activity.id}><Activity activity={activity} /></li>
      ))}
    </ul>
  );
}
