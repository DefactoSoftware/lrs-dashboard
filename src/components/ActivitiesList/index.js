import React from 'react';
import Activity from '../Activity';
import CSS from './style.css';

export default ({ activities })=> {
  return (
    <ul className={CSS.ActivitiesList}>
      {activities.map(activity => (
        <li className={CSS.item} key={activity.id}><Activity activity={activity} /></li>
      ))}
    </ul>
  );
}
