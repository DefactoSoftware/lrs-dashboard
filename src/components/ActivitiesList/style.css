.ActivitiesList {
  margin: 0 auto;
}

.item {
  border-bottom: 1px solid #ccc;
}
