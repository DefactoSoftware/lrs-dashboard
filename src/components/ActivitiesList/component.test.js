import React from 'react';
import ActivitiesList from './component';
import renderer from 'react-test-renderer';

it('renders no activities when no activities are present', () => {
  const activities = [];

  const tree = renderer.create(<ActivitiesList activities={activities} />).toJSON();

  expect(tree).toMatchSnapshot();
});


it('renders three activities when three activities are present', () => {
  const activities = [
    {
      id: "1234",
      actor: { mbox: 'foo@bar.com' },
      verb: { display: { 'en-US': 'created' } },
      object: { definition: { name: { 'en-US': 'Chapter' } } },
    },
    {
      id: "5768",
      actor: { mbox: 'bar@foo.com' },
      verb: { display: { 'en-US': 'updated' } },
      object: { definition: { name: { 'en-US': 'Space' } } },
    },
    {
      id: "9012",
      actor: { mbox: 'foo@bar.com' },
      verb: { display: { 'en-US': 'created' } },
      object: { definition: { name: { 'en-US': 'Portrait' } } },
    },
  ];

  const tree = renderer.create(<ActivitiesList activities={activities} />).toJSON();

  expect(tree).toMatchSnapshot();
});
