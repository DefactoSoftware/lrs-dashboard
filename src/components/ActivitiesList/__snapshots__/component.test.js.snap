exports[`test renders no activities when no activities are present 1`] = `
<ul
  className="ActivitiesList" />
`;

exports[`test renders three activities when three activities are present 1`] = `
<ul
  className="ActivitiesList">
  <li
    className="item">
    <p>
      <span>
        foo@bar.com
      </span>
       
      <strong>
        created
      </strong>
       
      <span>
        Chapter
      </span>
    </p>
  </li>
  <li
    className="item">
    <p>
      <span>
        bar@foo.com
      </span>
       
      <strong>
        updated
      </strong>
       
      <span>
        Space
      </span>
    </p>
  </li>
  <li
    className="item">
    <p>
      <span>
        foo@bar.com
      </span>
       
      <strong>
        created
      </strong>
       
      <span>
        Portrait
      </span>
    </p>
  </li>
</ul>
`;
